src/main/java/com/aws/greengrass/util/S3SdkClientFactory.java

```diff
@@ -88,4 +95,30 @@ public S3Client getClientForRegion(Region r) {
                 .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(true).build())
                 .credentialsProvider(credentialsProvider).region(r).build());
     }
+
+    /**
+     * Set s3 endpoint type.
+     *
+     * @param type s3EndpointType
+     */
+    private void setS3EndpointType(String type) {
+        //Check if system property and device config are consistent
+        //If not consistent, set system property according to device config value
+        String s3EndpointSystemProp = System.getProperty(S3_ENDPOINT_PROP_NAME);
+        boolean isGlobal = S3EndpointType.GLOBAL.name().equals(type);
+
+        if (isGlobal && S3_REGIONAL_ENDPOINT_VALUE.equals(s3EndpointSystemProp)) {
+            System.clearProperty(S3_ENDPOINT_PROP_NAME);
+            refreshClientCache();
+            logger.atInfo().log("s3 endpoint set to global");
```

Comment:
do not log this at info, this is going to happen a lot

Reply:
Updated